src/components/standalone/security/ips/IpsCreateBypassDrawer.vue

```diff
@@ -1,22 +1,137 @@
 <script lang="ts" setup>
-import { NeSideDrawer } from '@nethesis/vue-components'
+import { NeButton, NeRadioSelection, NeSideDrawer, NeTextInput } from '@nethesis/vue-components'
 import { useI18n } from 'vue-i18n'
+import { ref, watch } from 'vue'
+import type { AddressType, ByPass, Direction } from '@/composables/useIps'
+import { ubusCall, ValidationError } from '@/lib/standalone/ubus'
+import { MessageBag } from '@/lib/validation'
 
 const { t } = useI18n()
 
 const { visible = false } = defineProps<{
   visible: boolean
 }>()
 
-defineEmits(['close'])
+const emit = defineEmits<{
+  close: [void]
+  saved: [value: ByPass]
+}>()
+
+// FIXME: export the type from the library
+type RadioOption = {
+  label: string
+  id: string
+}
+
+watch(
+  () => visible,
+  (value) => {
+    if (value) {
+      protocol.value = 'ipv4'
+      direction.value = 'src'
+      ip.value = ''
+      description.value = ''
+      validationErrors.value.clear()
+    }
+  }
+)
+
+const protocolOptions: RadioOption[] = [
+  { label: 'IPv4', id: 'ipv4' },
+  { label: 'IPv6', id: 'ipv6' }
+]
+
+const directionOptions: RadioOption[] = [
+  { label: t('standalone.ips.source_bypass'), id: 'src' },
+  { label: t('standalone.ips.destination_bypass'), id: 'dst' }
+]
+
+const protocol = ref<AddressType>('ipv4')
+const direction = ref<Direction>('src')
+const ip = ref('')
+const description = ref('')
+const validationErrors = ref(new MessageBag())
+
+const loading = ref(false)
+const error = ref<Error>()
+
+function save() {
+  loading.value = true
+  error.value = undefined
+  validationErrors.value.clear()
+  ubusCall('ns.snort', 'create-bypass', {
+    protocol: protocol.value,
+    direction: direction.value,
+    ip: ip.value,
+    description: description.value
+  })
+    .then(() => {
+      emit('saved', {
+        direction: direction.value,
+        protocol: protocol.value,
+        ip: ip.value,
+        description: description.value
+      })
+    })
+    .catch((reason: Error) => {
+      if (reason instanceof ValidationError) {
+        validationErrors.value = reason.errorBag
+      } else {
+        error.value = reason
+      }
+    })
+    .finally(() => {
+      loading.value = false
+    })
+}
+
+function closeHandler() {
+  if (!loading.value) {
```

Comment:
Why this `if`? It should always be possible to close the drawer, to keep UI responsive

Reply:
Cause if you submit and close the handler while it's still sending the request, if an error arises you'll lose all the UI changes and won't get what happened (even a validation error)